### PR TITLE
Improve godoc; add synopsis

### PR DESCRIPTION
**Pull request by kc1212**
Documentation improvement on all packages

### CODE REVIEW

File: evoting/app/app.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// This is command line interface for communicating with the evoting service.
```
Comment (ineiti) [inline evoting/app/app.go]: s/is command/is a command/

File: ftcosi/README.md [hunk 2]
```diff
@@ -40,6 +40,43 @@ etc.) are not yet handled.
 The purpose of the project is to test scalability and robustness of this
 service on a testbed and to have a well-documented reusable code for it.
 
+## Implementation
+The protocol has four messages: 
+* Announcement which is sent from the root down the tree and announce the
```
Comment (ineiti) [inline ftcosi/README.md]: s/\*/-/

File: ftcosi/README.md [hunk 3]
```diff
@@ -40,6 +40,43 @@ etc.) are not yet handled.
...
+
+```
+       root
+         |
+         |
+     sub-leader
+      /       \
+     /         \
+  leaf_1 ... leaf_n
+```
```
Comment (ineiti) [inline ftcosi/README.md]: We need the same for `CoSi` and link to there from here. Or write it here and the copy it to the cosi-protocol.

Comment (kc1212) [reply ftcosi/README.md]: Added a section on implementation in cosi.

**Pull request merged**